tests/devices/unit_tests/test_utils.py

```diff
@@ -35,15 +39,32 @@ def test_full_status_gives_error_if_intermediate_status_fails():
 
 def test_check_call_back_error_gives_correct_error():
     LOGGER.error = MagicMock()
-    run_functions_without_blocking([get_bad_status])
 
-    run_functions_without_blocking([get_good_status])
-    LOGGER.error.assert_called_once()
+    with pytest.raises(StatusException):
+        returned_status = run_functions_without_blocking([get_bad_status])
+        returned_status.wait(0.1)
+
+    assert isinstance(returned_status.exception(), StatusException)
+
+    LOGGER.error.assert_called()
 
 
 def test_wrap_function_callback():
-    dummy_func = MagicMock(return_value=Status)
-    run_functions_without_blocking([lambda: get_good_status(), dummy_func])
+    dummy_func = MagicMock(return_value=Status())
+    returned_status = run_functions_without_blocking(
+        [lambda: get_good_status(), dummy_func]
+    )
+    dummy_func.assert_called_once
+    try:
+        returned_status.wait(0.1)
+    except BaseException:
+        pass
+
+
+def test_wrap_function_callback_errors_on_wrong_return_type():
+    dummy_func = MagicMock(return_value=3)
+    with pytest.raises(ValueError):
+        run_functions_without_blocking([lambda: get_good_status(), dummy_func])
```

Comment:
Should: We should get the returned status and wait on it to make sure it's tidied